### PR TITLE
Introduce global and local generator attributes

### PR DESCRIPTION
**Pull request by mbryzek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -189,14 +189,45 @@ Example File w/ Settings:
...
+Example File w/ Local Generator Attributes:
+
+    code:
+      happycorp:
+        api-salary-calculator:
+            version: 1.3.5
+            generators:
+              - generator: play_2_6_client
+                target: src/test/generated
+                attributes:
```
Comment (gheine) [inline README.md]: should be "attributes" or "generator_attributes" on all levels

Comment (mbryzek) [reply README.md]: hmm... top level wanted it to be clear that we are settings attributes for generators. Here context is clear it is a generator.

Alternatively, maybe cleaner if we replace `generator_attributes` w/
```
 attributes:
   generators:
       ...
```
then we can keep things consistent

Comment (mbryzek) [reply README.md]: I feel like the attributes are specific to each generator - ie. joda time only makes sense in the context of the play generators supporting joda time attribute.

In future, we can add wildcard support if needed, eg:
```
 attributes:
   generators:
       "*"
         foo: bar
```

Comment (mbryzek) [reply README.md]: imagine 2 generators that both define an attribute named "foo" - it is entirely possible for the values to be incompatible and that would make it impossible to use the global settings.

I prefer to err on 100% correct here (even though it is unlikely we'll see a conflict)

**Comment by mbryzek** [comment]
@gheine ok to merge?

File: README.md (hunk 1 above)
Comment (gheine) [reply README.md]: Fair point. You could argue though that the generator keys should always be specific enough that there are no conflicts. And even if there should be, the attributes can always be overridden in the generator specific sections of the config file? The downside of doing it this way is that if you're using multiple generators like at Flow (play models, play json, client, routes, mock client, etc.) the attributes would need to be repeated in the config file making this section very verbose.  ¯\_(ツ)_/¯

**Comment by mbryzek** [comment]
@gheine FYI: added the wildcard attributes

**Pull request merged**